### PR TITLE
[SCX-326] Add Granite::Action::subject? helper method

### PR DESCRIPTION
**Pull request by aitorlb**
This PR responds to the following need:
> I wish there was an easy way to check if a given action defines a `subject` or not, without having to look into the class internals.
```ruby
# before
Action._subject.blank?
# after
Action.subject?
```

### Review

- [ ] Document code according to [Getting Started with Yard](http://www.rubydoc.info/gems/yard/file/docs/GettingStarted.md).
- [x] All tests are passing.
- [x] Test manually.
- [ ] Get approval.

### Pre-merge checklist

- [ ] The PR relates to a single subject with a clear title and description in grammatically correct, complete sentences.
- [ ] Verify that feature branch is up-to-date with `master` (if not - rebase it).
- [ ] Double check the quality of [commit messages](http://chris.beams.io/posts/git-commit/).
- [ ] Squash related commits together.

### CODE REVIEW

File: lib/granite/action/subject.rb [hunk 1]
```diff
@@ -48,6 +52,10 @@ def initialize(*args)
 
       private
 
+      def find_subject_reflection
+        self.class.reflect_on_association(self.class._subject)
+      end
```
Comment (aitorlb) [inline lib/granite/action/subject.rb:57]: This unrelated change can be undone. IMO, line 44 was inconsistent with the level of abstraction maintained in the rest of the `#initialize` method body.

Comment (andriusch) [reply lib/granite/action/subject.rb:57]: That's fine, but you have instance method than only calls 2 class methods. This should be a class method. Also let's rename this to `subject_reflection`.

File: spec/lib/granite/action/subject_spec.rb [hunk 2]
```diff
@@ -32,4 +32,18 @@
     specify { expect(Action.new(comment: 'Comment', subject: student).comment).to eq('Comment') }
     specify { expect(Action.new(student, comment: 'Comment').comment).to eq('Comment') }
   end
+
+  describe '::subject?' do
```
Comment (andriusch) [inline spec/lib/granite/action/subject_spec.rb:36]:
```suggestion
  describe '.subject?' do
```

File: spec/lib/granite/action/subject_spec.rb [hunk 3]
```diff
@@ -32,4 +32,18 @@
...
+    specify { expect(Action).to be_subject }
+
+    context 'when Action does not define subject' do
+      before do
+        stub_class(:action, Granite::Action) do
+          attribute :comment, String
+        end
```
Comment (andriusch) [inline spec/lib/granite/action/subject_spec.rb:43]: This is adding extra information to the spec but is not required for it. I think the best way to handle this would be:
1. Move `stub_class` in `describe Granite::Action::Subject` to `describe '#initialize'`
2. Have a very simple `stub_class` inside `describe '.subject?'` that only creates a class.
3. Change context to "when action defined subject", and have something like `before { Action.subject :student }`

File: lib/granite/action/subject.rb [hunk 4]
```diff
@@ -29,15 +29,19 @@ def subject(name, *args, &block)
...
+          _subject.present?
+        end
       end
 
       def initialize(*args)
-        if self.class._subject.blank?
+        unless self.class.subject?
           super
           return
         end
```
Comment (andriusch) [inline lib/granite/action/subject.rb:42]: Not strictly necessary since it's not your code, but I think it would be better to change this to
```suggestion
        return super unless self.class.subject?
```

**Review by andriusch — CHANGES_REQUESTED** [review]

File: lib/granite/action/subject.rb (hunk 4 above)
Comment (aitorlb) [reply lib/granite/action/subject.rb:42]: :+1:

Comment (aitorlb) [reply lib/granite/action/subject.rb:42]: Need to disable rubocop:
```ruby
return super unless self.class.subject? # rubocop:disable Lint/ReturnInVoidContext
```

File: spec/lib/granite/action/subject_spec.rb (hunk 3 above)
Comment (aitorlb) [reply spec/lib/granite/action/subject_spec.rb:43]: :+1:

File: spec/lib/granite/action/subject_spec.rb (hunk 2 above)
Comment (aitorlb) [reply spec/lib/granite/action/subject_spec.rb:36]: :+1:

File: lib/granite/action/subject.rb (hunk 1 above)
Comment (aitorlb) [reply lib/granite/action/subject.rb]: :+1:

File: spec/lib/granite/action/subject_spec.rb [hunk 5]
```diff
@@ -32,4 +32,18 @@
...
+  describe '.subject?' do
+    before { stub_class(:action, Granite::Action) }
+
+    context 'when action defines subject' do
+      before { Action.subject(:student) }
+
+      specify { expect(Action).to be_subject }
+    end
+
+    context 'when action does not define subject' do
```
Comment (andriusch) [inline spec/lib/granite/action/subject_spec.rb:45]: Don't need this context, just move the spec to `describe '.subject?'`

**Review by andriusch — APPROVED** [review]
